.circleci/config.yml

```diff
@@ -887,48 +896,106 @@ workflows:
     jobs:
       - test_profile:
           <<: *default-context
+
+  # This workflow can be triggered manually on the PR
   react-17:
+    when:
+      equal: [react-17, << pipeline.parameters.workflow >>]
+    jobs:
+      - test_unit:
+          <<: *default-context
+          react-version: ^17.0.0
+          name: test_unit-react@17
+      - test_browser:
+          <<: *default-context
+          react-version: ^17.0.0
+          name: test_browser-react@17
+      - test_regressions:
+          <<: *default-context
+          react-version: ^17.0.0
+          name: test_regressions-react@17
+      - test_e2e:
+          <<: *default-context
+          react-version: ^17.0.0
+          name: test_e2e-react@17
+
+  # This workflow is identical to react-17, but scheduled
+  react-17-cron:
     triggers:
       - schedule:
           cron: '0 0 * * *'
           filters:
             branches:
               only:
                 - master
+                - next
     jobs:
       - test_unit:
           <<: *default-context
           react-version: ^17.0.0
+          name: test_unit-react@17
       - test_browser:
           <<: *default-context
           react-version: ^17.0.0
+          name: test_browser-react@17
       - test_regressions:
           <<: *default-context
           react-version: ^17.0.0
+          name: test_regressions-react@17
       - test_e2e:
           <<: *default-context
           react-version: ^17.0.0
+          name: test_e2e-react@17
+
+  # This workflow can be triggered manually on the PR
   react-next:
+    when:
+      equal: [react-next, << pipeline.parameters.workflow >>]
+    jobs:
+      - test_unit:
+          <<: *default-context
+          react-version: next
+          name: test_unit-react@next
```

Comment:
Added names here to avoid confusing job names, like this:

<img width="562" src="https://github.com/mui/material-ui/assets/13808724/88eac449-ccaa-40ae-95c2-f322e04dccdc">

Reply:
I'm not sure why these ghost jobs appear though:

<img width="858" src="https://github.com/user-attachments/assets/22c3506a-aa75-4e1c-9924-8854d9232af8">

Perhaps it's a lag on the CircleCI side.

Reply:
Maybe it got in a bad state after renaming those jobs? Maybe try opening a different PR?

Reply:
Tried in https://github.com/mui/material-ui/pull/42991 and the result is the same...

Reply:
I believe circleci reuses some things when opening a R for the same commit. At least I think I've observed that before. Trying out that hypothesis in https://github.com/mui/mui-x/pull/13942

Reply:
Thanks, that seems to work!
I'll trigger the `react-next` workflow to make sure everything works as expected.

Reply:
All looks good now!
Any objections to merging current PR?

Reply:
No, go for it 👍

Reply:
Seeing these empty CI jobs after merging this in #42990 and other latest PRs. What can I do?